ntoskrnl/io/iomgr/deviface.c

```diff
@@ -30,185 +31,450 @@ static PWCHAR BaseKeyString = L"\\Registry\\Machine\\System\\CurrentControlSet\\
 
 static
 NTSTATUS
-OpenRegistryHandlesFromSymbolicLink(IN PUNICODE_STRING SymbolicLinkName,
-                                    IN ACCESS_MASK DesiredAccess,
-                                    IN OPTIONAL PHANDLE GuidKey,
-                                    IN OPTIONAL PHANDLE DeviceKey,
-                                    IN OPTIONAL PHANDLE InstanceKey)
+IopBuildSymbolicLink(IN PUNICODE_STRING PrefixString,
+                     IN PUNICODE_STRING DeviceString,
+                     IN PUNICODE_STRING GuidString,
+                     IN OPTIONAL PUNICODE_STRING ReferenceString,
```

Comment:
This is not optional the way you're using it. Optional would mean that `ReferenceString` can be `NULL`, but you only allow `ReferenceString.Buffer` to be NULL, which is valid for any UNICODE_STRING (if `Length` is 0)

Reply:
OK, fixed this below.